tap_gitlab/__init__.py

```diff
@@ -87,6 +91,26 @@ def load_schema(entity):
         'key_properties': ['project_id', 'merge_request_iid', 'commit_id'],
         'replication_method': 'FULL_TABLE',
     },
+    'merge_request_detail': {
+        'url': '/projects/{project_id}/merge_requests/{merge_request_id}',
+        'schema': load_schema('merge_requests'),
+        'key_properties': ['id'],
+        'replication_method': 'INCREMENTAL',
+        'replication_keys': ['updated_at'],
+    },
+    'merge_request_approvals': {
+        'url': '/projects/{id}/merge_requests/{secondary_id}/approvals',
+        'schema': load_schema('merge_request_approvals'),
+        'key_properties': ['id'],
+        'replication_method': 'INCREMENTAL',
+        'replication_keys': ['updated_at'],
+    },
+    'merge_request_discussions': {
+        'url': '/projects/{id}/merge_requests/{secondary_id}/discussions',
+        'schema': load_schema('merge_request_discussions'),
+        'key_properties': ['id'],
+        'replication_method': 'FULL_TABLE',
```

Comment:
@raulgzm Why are we storing the merge_request_discussions information on each execution instead of performing it incrementally?

Reply:
I don't understand the code of the Tap completely, but I think that it is because the data returned for the API does not have a date-time field to compare, filter, and do that incremental process. If you take a look at the response from the API it is a list of discussions with just an ID and a list of notes. 

That it's the reason to get all the list at each execution. 

https://docs.gitlab.com/ee/api/discussions.html

Reply:
Understood, I was wondering if using the updated_at of the last comment could work for us, but maybe it doesn't worth the effort... the current approach it's good from my side

Reply:
It's not a bad idea, but it's something we'd need to manage through code completely because the API doesn't allow us to filter through that field. We will need to process all the items from the API, but it's true we will write just the new ones in the DB. Is that what you had in mind? 

Maybe it is something to note for the future.